openedx/core/djangoapps/enrollments/services.py

```diff
@@ -1,20 +1,78 @@
 """
 Enrollments Service
 """
+from functools import reduce
+from operator import or_
 
+from django.conf import settings
 
+from common.djangoapps.course_modes.models import CourseMode
 from common.djangoapps.student.models import CourseEnrollment
-
+from django.db.models import Q
+from opaque_keys.edx.keys import CourseKey
+from xmodule.modulestore.django import modulestore
 
 class EnrollmentsService(object):
     """
     Enrollments service
 
     Provides functions related to course enrollments
     """
-
     def get_active_enrollments_by_course(self, course_id):
         """
         Returns a list of active enrollments for a course
         """
-        return list(CourseEnrollment.objects.filter(course_id=course_id, is_active=True))
+        return CourseEnrollment.objects.filter(course_id=course_id, is_active=True)
+
+    def _get_enrollments_for_course_proctoring_eligible_modes(self, course_id, allow_honor_mode):
```

Comment:
I would say `allow_honor_mode` parameter can be default to `False` here. This way, it can be easily removed in the future.